percy/snapshot.py

```diff
@@ -1,74 +1,62 @@
 import os
+import platform
 import requests
 
-from selenium import webdriver
+from selenium.webdriver import __version__ as SELENIUM_VERSION
+from percy.version import __version__ as SDK_VERSION
 
-AGENT_URL = 'http://localhost:5338'
-VERSION = 'v0.1.2'
+# for logging
+LABEL = '[\u001b[35mpercy\u001b[39m] '
 
-percyIsRunning = True
+# Collect client and environment information
+CLIENT_INFO = 'percy-selenium-python/' + SDK_VERSION
+ENV_INFO = ['selenium/' + SELENIUM_VERSION, 'python/' + platform.python_version()]
 
-# Fetches the JS that serializes the DOM
-def getAgentJS():
-    global percyIsRunning
+# Maybe get the CLI API address from the environment
+PERCY_CLI_API = os.environ.get('PERCY_CLI_API') or 'http://localhost:5338/percy'
+PERCY_LOGLEVEL = os.environ.get('PERCY_LOGLEVEL') or 'info'
 
-    try:
-        agentJS = requests.get(AGENT_URL + '/percy-agent.js')
-        return agentJS.text
-    except requests.exceptions.RequestException as e:
-        if isDebug():
-            print(e)
-        if percyIsRunning == True:
-            percyIsRunning = False
-        print('[percy] failed to fetch percy-agent.js, disabling Percy')
-        return percyIsRunning
-
-
-# POSTs the serialized DOM to the percy-agent server for asset discovery
-def postSnapshot(postData):
-    try:
-        requests.post(AGENT_URL + '/percy/snapshot', json=postData)
-    except requests.exceptions.RequestException as e:
-        if isDebug():
-            print(e)
+# Cache @percy/dom script to avoid extraneous requests
+PERCY_DOM_SCRIPT = None
 
-        print('[percy] failed to POST snapshot to percy-agent:' + postData.get('name'))
-        return
+# Check if Percy is enabled while caching the @percy/dom script
+def isPercyEnabled():
+    global PERCY_DOM_SCRIPT
 
-def clientInfo():
-    return 'percy-selenium-python/' + VERSION
+    if PERCY_DOM_SCRIPT is None:
+        try:
+            r = requests.get(PERCY_CLI_API + '/dom.js')
+            r.raise_for_status()
+            PERCY_DOM_SCRIPT = r.text
+        except Exception as e:
+            if PERCY_LOGLEVEL == 'debug': print(e)
+            PERCY_DOM_SCRIPT = ''
```

Comment:
Semantic difference between empty string and None is... well it's understandable, but it will trip however reads the code next.

Given that `isPercyEnabled` is used exactly once, how about hiding the caching logic behind e.g. `functools.cache` and thus removing the magic `None vs. ""` distinction?

Reply:
Thanks for the tip @dimaqq! We're not as familiar with Python as we'd like to be, so the pointer towards `functools.cache` is great! I've added another commit that incorporates it along with separating the healthcheck from fetching this script.